test/fabscript

```diff
@@ -21,32 +21,32 @@ src = module('..src')
 python_libs=python.instance().libs
 features |= runpath(src.bpl.path, base='')
 
-def extension_test(name, ext=[], script=None, np=False,
+def extension_test(name, exts=[], script=None, numpy=False,
                    features=features, condition=None):
     """Create a Python extension test `name`.
     Arguments:
       * name: the name of the test.
-      * ext: extensions to be compiled, <name> if none are given.
+      * exts: extensions to be compiled, <name> if none are given.
       * script: the test script to execute, <name>.py if none is given.
-      * np: if true, add boost_numpy to sources
+      * numpy: if true, add boost_numpy to sources
       * features: pre-defined features
       * condition: any condition under which to run the test
     Return:
       * the test artefact"""
 
     features=features.copy()
     extensions = []
-    libs = [src.bnl, src.bpl] if np else [src.bpl]
-    for e in ext or [name]:
-        if type(e) is str:  # build from a single source file
-            n = e if e != name else e + '_ext'
-            s = [e + '.cpp']
+    libs = [src.bnl, src.bpl] if numpy else [src.bpl]
+    for ext in exts or [name]:
+        if type(ext) is str:  # build from a single source file
+            ext_name = ext if ext != name else ext + '_ext'
+            sources = [ext + '.cpp']
         else:  # build from a list of source files
-            n = e[0] if e[0] != name else e[0] + '_ext'
-            s = [n + '.cpp' for n in e]
```

Comment:
In Python 2, this `n` will leak out of the generator and trample over the previous value.

Reply:
Great finding ! Good that Python 2 has reached its end of life. I'm about to remove Python 2 support from Faber, too...